core/src/test/java/hudson/UtilTest.java

```diff
@@ -490,9 +494,13 @@ private void lockFileForDeletion(File f) throws IOException, InterruptedExceptio
         // On unix, can't use "chattr +u" because ext fs ignores it.
         // On Windows, we can't delete files that are open for reading, so we use that.
         assert Functions.isWindows();
-        final InputStream s = new FileInputStream(f);
+        final FileChannel channel = FileChannel.open(f.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
+        final FileLock lock = channel.lock();
```

Comment:
How is this *worse* than the code that was there before? 

The point of code review is not to get the reviewed code perfect, but to ensure that the code quality is an increasing function. i.e. `dQ/dC < 0` is grounds to put a bug but `dQ/dC == 0` is not (though `dQ/dC > 0` is preferred obviously) 

If we keep the attitude that `dQ/dC == ∞` then we will never make any changes

(where `Q` is quality and `C` is changes to the code) 

With this code, if run on a windows filesystem that does not support locking we will get a test Error rather than a test Failure... which IMHO is actually the correct result... so to put the point further, I would argue that this is actually the correct behaviour from test code (although my intent had been like for like so this is more of an unintended virtuous side-effect ;-) )